src/test/java/io/orangebeard/listener/OrangebeardListenerTest.java

```diff
@@ -0,0 +1,172 @@
+        UUID testUuid21  = UUID.fromString("c4be0af9-8f0f-478a-a66b-5697be40407a");
+
+        String testName22 = "SubSuite 2.2";
+        Description testDescription22 = Description.createTestDescription(suiteName2, testName22);
+        UUID testUuid22  = UUID.fromString("b3650fc1-41bc-4b00-beec-8c4c1467cfd0");
+
+        suiteDescription1.addChild(testDescription11);
+        suiteDescription1.addChild(testDescription12);
+
+        suiteDescription2.addChild(testDescription21);
+        suiteDescription2.addChild(testDescription22);
+
+        // Prepare the UUID's that orangebeardClient.startTestItem(...) should return.
+        when(orangebeardClient.startTestItem(any(), any(StartTestItem.class)))
+                .thenReturn(suiteUuid1)
+                .thenReturn(testUuid11)
+                .thenReturn(testUuid12)
+                .thenReturn(suiteUuid2)
+                .thenReturn(testUuid21)
+                .thenReturn(testUuid22);
+
+        // Prepare to capture the parameters passed to orangebeardClient.startTestItem(...).
+        ArgumentCaptor<StartTestItem> testItemCaptor = ArgumentCaptor.forClass(StartTestItem.class);
+        ArgumentCaptor<UUID> suiteUuidCaptor = ArgumentCaptor.forClass(UUID.class);
+
+
+        // Start the run, the suites, and the tests contained in the suites.
+        orangebeardListener.testRunStarted(runDescription);
+        orangebeardListener.testSuiteStarted(suiteDescription1);
+        orangebeardListener.testStarted(testDescription11);
+        orangebeardListener.testStarted(testDescription12);
+        orangebeardListener.testSuiteStarted(suiteDescription2);
+        orangebeardListener.testStarted(testDescription21);
+        orangebeardListener.testStarted(testDescription22);
+
+        // Verify the calls, and capture the parameters that were passed.
+        verify(orangebeardClient, times(6)).startTestItem(suiteUuidCaptor.capture(), testItemCaptor.capture());
+
+        // Verify that the tests were put in the proper suites.
+        List<StartTestItem> allStartTestItems = testItemCaptor.getAllValues();
+        List<UUID> allUuids = suiteUuidCaptor.getAllValues();
+        assertNull(allUuids.get(0));
+        assertEquals(suiteUuid1, allUuids.get(1));
+        assertEquals(suiteUuid1, allUuids.get(2));
+        assertNull(allUuids.get(3));
+        assertEquals(suiteUuid2, allUuids.get(4));
+        assertEquals(suiteUuid2, allUuids.get(5));
+
+        // Verify that the tests were registered with the right names.
+        assertEquals(suiteName1, allStartTestItems.get(0).getName());
+        assertEquals(testName11, allStartTestItems.get(1).getName());
+        assertEquals(testName12, allStartTestItems.get(2).getName());
+
+        assertEquals(suiteName2, allStartTestItems.get(3).getName());
+        assertEquals(testName21, allStartTestItems.get(4).getName());
+        assertEquals(testName22, allStartTestItems.get(5).getName());
+
+    }
+
+}
```

Comment:
Can you upgrade to the latest Java client in the pom as well? I've fixed the NPE Bas found earlier in the latest release.

Reply:
I've set it to `<java.client.version>1.1.19-SNAPSHOT</java.client.version>`.
This works locally, after I installed that version of the listener.
But it fails to build on CI :
```
Error:  Failed to execute goal on project junit4-listener: Could not resolve dependencies for project io.orangebeard:junit4-listener:jar:0.0.7-SNAPSHOT: Could not find artifact io.orangebeard:java-client:jar:1.1.19-SNAPSHOT -> [Help 1]
```

Which version of the client should I use?